extensions/resource-deployment/src/ui/modelViewUtils.ts

```diff
@@ -390,6 +396,35 @@ async function hookUpDynamicEnablement(context: WizardPageContext): Promise<void
 	}));
 }
 
+async function hookUpValueProviders(context: WizardPageContext): Promise<void> {
+	await Promise.all(context.pageInfo.sections.map(async section => {
+		if (!section.fields) {
+			return;
+		}
+		await Promise.all(section.fields.map(async field => {
+			if (field.valueProvider) {
+				const fieldKey = field.variableName || field.label;
+				const fieldComponent = context.inputComponents[fieldKey];
+				const targetComponent = context.inputComponents[field.valueProvider.triggerField];
+				if (!targetComponent) {
+					console.error(`Could not find target component ${field.valueProvider.triggerField} when hooking up value providers for ${field.label}`);
+					return;
+				}
+				const provider = valueProviderService.getValueProvider(field.valueProvider.providerId);
+				const updateFields = async () => {
+					const targetComponentValue = await targetComponent.getValue();
+					const newFieldValue = await provider.getValue(targetComponentValue?.toString() ?? '');
+					fieldComponent.setValue(newFieldValue);
```

Comment:
I think we have event listening to the targetComponet's value changed event, and this will trigger the model to be updated, right?

Reply:
Not sure what you mean by this

Reply:
for your scenario, will the tenent information be updated in the model or is this only updating the inputbox, i think input box already have events hooked up to update the model, but not 100% sure.

Reply:
That's done right before generating the notebook and goes through and calls getValue to get the value of each component : https://github.com/microsoft/azuredatastudio/blob/d2711a36d5cb34bc9fc331c69c3a0a29d3bc88a1/extensions%2Fresource-deployment%2Fsrc%2Fui%2FmodelViewUtils.ts#L1426

So we don't have to update anything in real-time, it'll get the correct value when it's all done.